### PR TITLE
CB-9436 Refactor browserify to get rid of `require-tr` transform

### PR DESCRIPTION
**Pull request by vladimir-kotikov**
This PR removes excess `require-tr` browserify transform and makes resultant bundle compatible with mobilespec and probably most of plugins.

NOTE: This PR depends on https://github.com/apache/cordova-lib/pull/275

### CODE REVIEW

File: tasks/lib/packager-browserify.js [hunk 1]
```diff
@@ -46,25 +37,13 @@ module.exports = function generate(platform, useWindowsLineEndings, platformVers
...
-        libraryRelease.pipeline.get('wrap').push(addSymbolList);
-
         // write license header
         writeLicenseHeader(outReleaseFileStream, platform, commitId, platformVersion);
 
-        releaseBundle = libraryRelease.bundle();
-
-        releaseBundle.pipe(outReleaseFileStream);
+        bundle(platform, false, commitId, platformVersion, platformPath)
+          .add(path.resolve(__dirname, '..', '..', 'src/scripts/bootstrap.js'))
```
Comment (stevengill) [inline tasks/lib/packager-browserify.js]: You add bootstrap.js here which has require('cordova/init') in it. But you also manually add require('cordova/init') in the form of init.js when adding plugins using cordova prepare --browserify. https://github.com/MSOpenTech/cordova-lib/commit/510516135ef41b735b2e610c1b8d95fc77bd571e#diff-28a16640f3807e6ee2300ea7ce8c2f1dR196

Are they both needed?

File: tasks/lib/bundle-browserify.js [hunk 2]
```diff
@@ -19,53 +19,63 @@
...
-    var b = browserify({debug: debug});
-    // XXX plugin_list is not present at this stage 
-    b.ignore(path.join(root, 'src', 'common', 'plugin_list'));
+module.exports = function bundle(platform, debug, commitId, platformVersion, platformPath) {
+    platformPath = fs.existsSync(platformPath) && fs.existsSync(path.join(platformPath, 'cordova-js-src')) ?
+        path.join(platformPath, 'cordova-js-src') :
+        path.resolve(root, 'src', 'legacy-exec', platform);
 
-    b.transform(require_tr.transform, {'platform': platform});
+    var platformDirname = platform === 'amazon-fireos' ? 'android' : platform;
```
Comment (stevengill) [inline tasks/lib/bundle-browserify.js]: hahaha smart

File: tasks/lib/packager-browserify.js (hunk 1 above)
Comment (vladimir-kotikov) [inline tasks/lib/packager-browserify.js]: Yes, they both needed. This was moved from `bundle-browserify.js` direcly to `cordova-lib/prepare-browserify` due to some reasons (see https://github.com/apache/cordova-lib/pull/275#issuecomment-127782310) but we still need to add this file when building clean cordova-js bundle (without plugins) from grunt task.

**Comment by stevengill** [comment]
Merge away! Thanks again. Love the refactor. Way simpler to read and follow.

**Pull request merged**